src/metabase/upload.clj

```diff
@@ -742,16 +739,15 @@
                                   (remove #(false? (:is_upload %)))
                                   (keep :table_id)
                                   set)
-        uploadable-table-ids (set (uploadable-table-ids table-ids))
-        based-on-upload      (fn [model]
-                               (when-let [dataset_query (:dataset_query model)] ; dataset_query is sometimes null in tests
-                                 (let [query (lib/->pMBQL dataset_query)]
-                                   (when (and (some-> model :query_type name (= "query"))
-                                              (contains? uploadable-table-ids (:table_id model))
-                                              (no-joins? query))
-                                     (lib/source-table-id query)))))]
-    (map #(m/assoc-some % :based_on_upload (based-on-upload %))
-         models)))
+        query?               (fn [model] (= "query" (name (:query_type model "query"))))
+        uploadable?          (comp (uploadable-table-ids table-ids) :table_id)]
```

Comment:
This function takes a model as an arg, yet doesn't determine whether a model is "uploadable" or not. A more appropriate name would be `has-uploadable-table-id?` but at that point you might as well inline it
```suggestion
        uploadable-table?          (comp (uploadable-table-ids table-ids) :table_id)]
```

Reply:
How about `uploadable-id?` I have a pretty strong aversion to mixing abstraction levels in the `(when (and x y z) ...)` cluster, a capital offense in my fashion guide.

Reply:
Or `uploadable-table`?

Reply:
`uploadable-id?` suffers from the same problem because the "id" in the name is ambiguous. `uploadable-table?` is pretty clear though.

For myself I don't have such a strong aversion to mixing abstration levels. This is all in the name of readability and introducing extra names comes at a cost because they can be misleading and will make your eyes jump around if you have to understand them.

Reply:
That `has-` hurts me - especially at it forces a re-indent on the let bindings, but agree that without it there's a bit more ambiguity though - `uploadable-table?` could make it sound like there is "exactly one" uploadable tables, when really its "at least one". We rely on the lack of joins to make it exact, and that call is more expensive so it makes sense to keep it later in the `and`.

Reply:
Yeah. To understand what the table in `uploadable-table?` means you almost need to know that it's accessing the table from the `source_table` from the first stage of the query. It's unfortunately complicated